File: engine/engine-model/src/main/scala/tools/packet/wedding/MarriageRequest.scala
package tools.packet.wedding

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class MarriageRequest(private var _name: String, private var _characterId: Int) extends PacketInput {
  def name: String = _name

  def characterId: Int = _characterId

  override def opcode(): SendOpcode = SendOpcode.MARRIAGE_REQUEST
}

File: engine/engine-model/src/main/java/tools/packet/wedding/MarriageRequest.java
package tools.packet.wedding;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record MarriageRequest(String name, Integer characterId) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.MARRIAGE_REQUEST;
   }
}